packages/typespec-python/src/emitter.ts

```diff
@@ -608,7 +608,7 @@ function addLroInformation(
     const lroMeta = getLroMetadata(context.program, tspOperation);
     if (!isAzureCoreModel(lroMeta!.logicalResult)) {
         emittedOperation["responses"][0]["type"] = getType(context, lroMeta!.logicalResult);
-        if (lroMeta!.finalStep?.target.kind === "ModelProperty" && emittedOperation.method !== "PUT") {
+        if (lroMeta!.finalStep?.target.kind === "ModelProperty") {
```

Comment:
https://github.com/Azure/autorest.python/pull/2040#discussion_r1287970412 is resolved by https://github.com/Azure/typespec-azure/pull/3403 so we could remove the hard code now.